landscape/lib/network.py

```diff
@@ -124,58 +124,93 @@ def get_default_interfaces():
     return interfaces
 
 
-def get_active_device_info(skipped_interfaces=("lo",),
-                           skip_vlan=True, skip_alias=True,
-                           extended=False, default_only=False):
+def get_filtered_if_info(filters=(), extended=False):
     """
-    Returns a dictionary containing information on each active network
-    interface (which can be filtered based on flags) present on a machine.
+    Returns a dictionary containing info on each active network
+    interface that passes all `filters`.
+
+    A filter is a callable that returns True if the interface should be
+    skipped.
     """
     results = []
-    default_interfaces = get_default_interfaces()
+
     try:
         sock = socket.socket(socket.AF_INET, socket.SOCK_DGRAM,
                              socket.IPPROTO_IP)
+
         for interface in netifaces.interfaces():
-            if interface in skipped_interfaces:
-                continue
-            if skip_vlan and "." in interface:
-                continue
-            if skip_alias and ":" in interface:
-                continue
-            if default_only and interface not in default_interfaces:
+            if any(f(interface) for f in filters):
                 continue
+
             ifaddresses = netifaces.ifaddresses(interface)
             if not is_active(ifaddresses):
                 continue
-            flags = get_flags(sock, interface.encode())
+
+            ifencoded = interface.encode()
+            flags = get_flags(sock, ifencoded)
             if not is_up(flags):
                 continue
-            interface_info = {"interface": interface}
-            interface_info["flags"] = flags
-            speed, duplex = get_network_interface_speed(
-                sock, interface.encode())
-            interface_info["speed"] = speed
-            interface_info["duplex"] = duplex
+
             ip_addresses = get_ip_addresses(ifaddresses)
+            if not (netifaces.AF_INET in ip_addresses or extended):
```

Comment:
Could be if not extended:
                    if AF_INET not in ip_adrresses:
                          continue

Reply:
de morganing it to `if not extended and netifaces.AF_INET not in ip_addresses` seems like a good compromise.